app/models/asset.rb

```diff
@@ -58,6 +66,54 @@ def dzi_file
     @dzi_file ||= DziFiles.new(self)
   end
 
+  # OVERRIDE of standard Kithe update index, to:
+  #   * update PARENT in solr on change, we don't index Assets, but do index
+  #     some aspects of assets in their parent Works.
+  #   * and only when allow-listed attributes we know we index on parent have changed
+  def update_index(mapper: kithe_indexable_mapper, writer:nil, **)
+    if should_reindex_parent_after_save?
+      # WEIRD workaround, in some cases the parent still this record in memory
+      # even though it's been, and will use that in-memory list of members for
+      # indexing in Solr! If that's going on, we need to make sure to reset
+      # the association.
+      if self.destroyed? && parent.members.loaded? && parent.members.include?(self)
```

Comment:
Meaning -- in some cases the parent work still contains this asset in its `members` association even though the child asset has already been deleted, and could attempt to index information from the deleted asset.